asset/src/_kafka_clients/consumer-client.ts

```diff
@@ -95,6 +135,19 @@ export default class ConsumerClient extends BaseClient<kafka.KafkaConsumer> {
         }
     }
 
+    async onRetry(): Promise<void> {
+        if (this.rollback_on_failure) {
+            await this.rollback();
+        } else {
+            this._logger.warn('committing kafka offsets on slice retry - THIS MAY CAUSE DATA LOSS');
+            await this.commit(this.use_commit_sync);
+        }
+    }
+
+    async onFinalizing(): Promise<void> {
```

Comment:
`onFinalizing` doesn't really make sense from the context of this file, if we changed `commit(useSync?: boolean)` to `commit()` - and then have it use the setting `this.useCommitSync` internally, I think it would be intuitive.